File: cpp-solutions/230.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
private: 
    vector<int> vec;
public:
    int kthSmallest(TreeNode* root, int k) {
        if(!root) return 0;
        inOrder(root, k);
        return vec.back();
    }
    
    void inOrder(TreeNode *root, int k){
         if(root->left){
             inOrder(root->left, k);
         }
         if(vec.size()==k) return;
         
         vec.push_back(root->val);
         if(vec.size()==k) return;
         
         if(root->right){
             inOrder(root->right, k);
         }
    }
};

File: golang/230/230.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

var cnt int
var res int

func kthSmallest(root *TreeNode, k int) int {
	cnt, res = 0, 0

	inorder(root, k)
	return res
}

func inorder(root *TreeNode, k int) {
	if root == nil {
		return
	}

	inorder(root.Left, k)
	cnt++
	if cnt == k {
		res = root.Val
		return
	}
	inorder(root.Right, k)
}
